### PR TITLE
Make Loki module path v2 compatible

### LINKED ISSUE #2826: v2 go module semantic versioning
**Describe the bug**
I'm using go client library (`/pkg/promtail/client`) instead of promtail to push log entries to Loki. 
After updating Loki to v2.0.0 I cannot use go client library in projects with `go.mod` because Loki's module path does not contain `/v2` suffix as [required by the go command](https://golang.org/cmd/go/#hdr-Module_compatibility_and_semantic_versioning).

> To preserve import compatibility, the go command requires that modules with major version v2 or later use a module path with that major version as the final element. For example, version v2.0.0 of example.com/m must instead use module path example.com/m/v2, and packages in that module would use that path as their import path prefix, as in example.com/m/v2/sub/pkg.

**To Reproduce**

- `mkdir myproject && cd myproject`
- `go mod init myproject`
- `go get github.com/grafana/loki@v2.0.0`

**Expected behavior**
Be able to use Loki v2 go client library in projects with `go.mod` file (module-aware mode).

**Environment:**
 local, linux

**Screenshots, Promtail config, or terminal output**
`go get github.com/grafana/loki@v2.0.0: github.com/grafana/loki@v2.0.0: invalid version: module contains a go.mod file, so major version must be compatible: should be v0 or v1, not v2`

### PR DESCRIPTION
**Pull request by kavirajk**
**What this PR does / why we need it**:
Make Loki version go module path compatible

**Which issue(s) this PR fixes**:
Fixes #2826 #3361 

**Special notes for your reviewer**:

**Checklist**
- [ ] Documentation added
- [ ] Tests updated

### CODE REVIEW

**Review by owen-d — COMMENTED** [review]
I wish we could alias `grafana/loki` -> `grafana/loki/v2` in order to avoid updating _every_  reference :/ . That being said, this will really help importing!

**Comment by kavirajk** [comment]
Yea I understand what you mean. I wish if there is any better way. Unfortunately I don’t think we can alias it.  :(

I think thats kinda contrary to the whole point of having version (/v2, /v3) explicitly in the import path.
for example, how would you use both v1 and v2 on the same file if we alias it?

I understand its kinda extra step every time we do major release (no change for any minor version releases though)
we can automate it via `mod`  tool (also recommended by Go wiki).
https://github.com/marwan-at-work/mod

so that when we go from v2 -> v3, we just have to run `mod upgrade`  and it will replace all the import path

**Comment by stefanhenseler** [comment]
@kavirajk Do you know when this will be merged? I'm using some packages from Loki in one of my projects and run into issues without this change. There are multiple PRs open about this, but no action so far.

**Comment by kavirajk** [comment]
@stefanhenseler totally understand your pain :( We have some large discussion #2826 

tl;dr is, even if we try to support go module path, it will be only from Loki v3 unfortunately!

**Comment by kavirajk** [comment]
Closing this as no activities for long time. Feel free to send new PR if anyone want's to revive the work
